submissions/nwokejeme-immaculeta/dom/signup.js

```diff
@@ -0,0 +1,187 @@
+        return true;
+    }
+}
+
+// password field validation
+function passwordvalidation(event) {
+    let pwdRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*(_|[^\w])).+$/;
+    if (!password.value.match(pwdRegex)) {
+    password.nextElementSibling.innerHTML = "Your password should have a minimum of 6 characters, 1 capital letter, 1 special character eg @ and 1 number.";
+    event.preventDefault();
+    return false;
+    }   
+        else if (!(password.value.length >= 6))  {
+        password.nextElementSibling.innerHTML = "Please length must be at least 6 characters";
+        event.preventDefault();   
+        return false;    
+    }
+        else  {
+        password.nextElementSibling.innerHTML = "";
+        return true;
+    }
+        
+}
+
+//confirm password field validation
+function cwordvalidation(event) {
+    let pwdRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*(_|[^\w])).+$/;
+    if (!confirmPassword.value.match(pwdRegex)) {
+    confirmPassword.nextElementSibling.innerHTML = "Please password should contain at least 1 letter, 1number and 1special character";
+    event.preventDefault();
+    return false;
+    }   
+        else if (!(confirmPassword.value.length >= 6))  {
+        confirmPassword.nextElementSibling.innerHTML = "Please length must be at least 6 characters";
+        event.preventDefault();   
+        return false;    
+    }
+        else if (confirmPassword.value != password.value)  {
+        confirmPassword.nextElementSibling.innerHTML = "do not match";
+        event.preventDefault();   
+        return false; 
+    }
+        else  {
+        confirmPassword.nextElementSibling.innerHTML = "";
+        return true;
+    }
+        
+}
+
+function validateForm(e){
+    required(firstName, e);
+    required(lastName, e);
+    required(otherName, e);
+    required(email, e);
+    required(password, e);
+    required(confirmPassword, e);
+    cwordvalidation(e)
+    nameValidation(e);
+    emailfieldvalidation(e);
+    phonefieldvalidation(e)
+    passwordvalidation(e);
+    return true;  
+}
+
+submitForm.addEventListener("submit", validateForm);
+
+
+//footer form validation
+let submitForm1 = document.getElementById('verifyForm'),
+    footerEmail = document.getElementById('footerEmail');
+
+    function footeremailvalidation(event) {
+        let emailRegex =  /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;;
+        if (!footerEmail.value.match(emailRegex)) {
+            footerEmail.nextElementSibling.innerHTML = "Please enter a valid email";
+            event.preventDefault();
+    
+        if (footerEmail.value == '')
+            footerEmail.nextElementSibling.innerHTML = "Please enter your Email Address";
+            event.preventDefault();
+            return false;
+        }   
+        
+        else {
+            footerEmail.nextElementSibling.innerHTML = "";
+            return true;
+        }
+    }
+
+    function verifyForm(e) {
```

Comment:
Is this validation meant to be a part of this PR?

Reply:
yes